File: backend/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace ELibrary.Models
{
    public class Book
    {
        [Key]
        public int Id { get; set; }
        public required string UserId { get; set; }
        public required string Content { get; set; }
        public required string TextAnalysis { get; set; }
        public required string GutenbergId { get; set; }
        public required string Title { get; set; }
        public required string Language { get; set; }
        public DateTime DownloadDate { get; set; }
    }
}

File: frontend/lib/domain/entities/book.dart
class Book {
  final String gutenbergId;
  final String content;
  final String textAnalysis;
  final String metadata;
  final String title;
  final String language;

  Book({
    required this.gutenbergId,
    required this.content,
    required this.textAnalysis,
    required this.metadata,
    required this.title,
    required this.language,
  });
}